flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/utils/SpecUtils.java

```diff
@@ -50,6 +50,7 @@ public static <T extends AbstractFlinkSpec> SpecWithMeta<T> deserializeSpecWithM
         try {
             ObjectNode wrapper = (ObjectNode) objectMapper.readTree(specWithMetaString);
             ObjectNode internalMeta = (ObjectNode) wrapper.remove(INTERNAL_METADATA_JSON_KEY);
+            internalMeta.remove("metadata");
```

Comment:
Good question! I've found the keywords can be quite confusing-- this is what the `SpecWithMeta` class looks like
```
public class SpecWithMeta<T extends AbstractFlinkSpec> {
    T spec;
    ReconciliationMetadata meta;
}
```
i.e. the object that contains this entire block:
```
{
  "apiVersion": "flink.apache.org/v1beta1",
  "metadata": {
    "generation": 5
  },
  "firstDeployment": false
}
```
despite metadata being in that keyword, there is nothing in that class that distinguishes
```
"metadata": {
    "generation": 5
  },
```
from the rest of the chunk of raw data returned from the CRD metadata (sorry for using metadata again). Therefore the SpecWithMeta object will still contain that unwanted data. I put that line because if not, the deserializer will yell when the `ReconciliationMetadata` object is instantiated, because it doesn't recognize that `metadata` block as one of its attributes (the `metadata` that we want to remove). 
<details><summary>nasty chunky error </summary>

```
{"instant":{"epochSecond":1716324562,"nanoOfSecond":547000000},"thread":"pool-5-thread-1","level":"ERROR","loggerName":"io.fabric8.kubernetes.client.extended.leaderelection.LeaderElector","message":"Exception during leader election","thrown":{"commonElementCount":0,"localizedMessage":"io.javaoperatorsdk.operator.OperatorException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","message":"io.javaoperatorsdk.operator.OperatorException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","name":"java.util.concurrent.CompletionException","cause":{"commonElementCount":4,"localizedMessage":"java.util.concurrent.ExecutionException: java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","message":"java.util.concurrent.ExecutionException: java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","name":"io.javaoperatorsdk.operator.OperatorException","cause":{"commonElementCount":4,"localizedMessage":"java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","message":"java.lang.IllegalStateException: com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","name":"java.util.concurrent.ExecutionException","cause":{"commonElementCount":3,"localizedMessage":"com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","message":"com.fasterxml.jackson.databind.JsonMappingException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","name":"java.lang.IllegalStateException","cause":{"commonElementCount":3,"localizedMessage":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","message":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"]) (through reference chain: org.apache.flink.kubernetes.operator.api.FlinkDeployment[\"status\"]->org.apache.flink.kubernetes.operator.api.status.FlinkDeploymentStatus[\"lifecycleState\"])","name":"com.fasterxml.jackson.databind.JsonMappingException","cause":{"commonElementCount":3,"localizedMessage":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])","message":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])","name":"java.lang.IllegalArgumentException","cause":{"commonElementCount":3,"localizedMessage":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])","message":"Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])","name":"com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException","extendedStackTrace":"com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])\n\tat com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1138) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2224) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1709) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1687) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:320) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4539) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4475) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.utils.SpecUtils.deserializeSpecWithMeta(SpecUtils.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.ReconciliationStatus.deserializeLastReconciledSpecWithMeta(ReconciliationStatus.java:72) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.ReconciliationStatus.deserializeLastReconciledSpec(ReconciliationStatus.java:60) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.CommonStatus.getLifecycleState(CommonStatus.java:80) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]\n\tat jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]\n\tat jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]\n\tat java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]\n\tat com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.model.jackson.BeanPropertyWriterDelegate.serializeAsField(BeanPropertyWriterDelegate.java:68) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.model.jackson.BeanPropertyWriterDelegate.serializeAsField(BeanPropertyWriterDelegate.java:68) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:479) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:318) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4719) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3964) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.client.utils.KubernetesSerialization.clone(KubernetesSerialization.java:360) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.api.config.ConfigurationService$1.clone(ConfigurationService.java:60) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.InformerManager.lambda$get$11(InformerManager.java:173) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat java.util.Optional.map(Unknown Source) ~[?:?]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.InformerManager.get(InformerManager.java:173) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.ManagedInformerEventSource.get(ManagedInformerEventSource.java:121) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.submitReconciliationExecution(EventProcessor.java:127) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.handleMarkedEventForResource(EventProcessor.java:119) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.handleAlreadyMarkedEvents(EventProcessor.java:415) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.start(EventProcessor.java:409) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.Controller.startEventProcessing(Controller.java:393) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.ControllerManager.lambda$startEventProcessing$4(ControllerManager.java:62) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.api.config.ExecutorServiceManager.lambda$executeAndWaitForAllToComplete$0(ExecutorServiceManager.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]\n"},"extendedStackTrace":"java.lang.IllegalArgumentException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])\n\tat com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4544) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4475) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.utils.SpecUtils.deserializeSpecWithMeta(SpecUtils.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.ReconciliationStatus.deserializeLastReconciledSpecWithMeta(ReconciliationStatus.java:72) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.ReconciliationStatus.deserializeLastReconciledSpec(ReconciliationStatus.java:60) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.status.CommonStatus.getLifecycleState(CommonStatus.java:80) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]\n\tat jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]\n\tat jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]\n\tat java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]\n\tat com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.model.jackson.BeanPropertyWriterDelegate.serializeAsField(BeanPropertyWriterDelegate.java:68) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.model.jackson.BeanPropertyWriterDelegate.serializeAsField(BeanPropertyWriterDelegate.java:68) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:772) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:479) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:318) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4719) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3964) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.fabric8.kubernetes.client.utils.KubernetesSerialization.clone(KubernetesSerialization.java:360) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.api.config.ConfigurationService$1.clone(ConfigurationService.java:60) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.InformerManager.lambda$get$11(InformerManager.java:173) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat java.util.Optional.map(Unknown Source) ~[?:?]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.InformerManager.get(InformerManager.java:173) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.source.informer.ManagedInformerEventSource.get(ManagedInformerEventSource.java:121) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.submitReconciliationExecution(EventProcessor.java:127) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.handleMarkedEventForResource(EventProcessor.java:119) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.handleAlreadyMarkedEvents(EventProcessor.java:415) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.event.EventProcessor.start(EventProcessor.java:409) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.processing.Controller.startEventProcessing(Controller.java:393) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.ControllerManager.lambda$startEventProcessing$4(ControllerManager.java:62) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat io.javaoperatorsdk.operator.api.config.ExecutorServiceManager.lambda$executeAndWaitForAllToComplete$0(ExecutorServiceManager.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]\nCaused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field \"metadata\" (class org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata), not marked as ignorable (2 known properties: \"firstDeployment\", \"apiVersion\"])\n at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: org.apache.flink.kubernetes.operator.api.reconciler.ReconciliationMetadata[\"metadata\"])\n\tat com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1138) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2224) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1709) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1687) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:320) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4539) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4475) ~[flink-kubernetes-operator-1.9-SNAPSHOT-shaded.jar:1.9-SNAPSHOT]\n\tat org.apache.flink.kubernetes.operator.api.utils.SpecUtils.deserializeSpecWithMeta(SpecUtils.java:61)
...
```

<img width="1512" alt="Screenshot 2024-05-30 at 4 11 53 PM" src="https://github.com/Shopify/flink-kubernetes-operator/assets/65053157/5a629d97-510d-4a2a-a991-06c48212a0cb">


</details>
does that answer your question? Please let me know if I've missed anything or if you think there's a better way to go about this.